### PR TITLE
Add Chinese 2-20

### PR DESCRIPTION
**Pull request by Osirisoo0O**

### CODE REVIEW

**Comment by stevewhims** [comment]
Thank you, @Osirisoo0O. Sorry about the long delay in getting to this. I'm investigating seeing whether I can have the changes reviewed, and then I can merge.

Thank you! :)
-Steve

File: bite-sized-cpp/posts/cn/002.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# #2. 头文件 (.h文件)
+
+如果说你想创建一个名为**MyClass**的类型。在C#中&mdash;如果这是一个不涉及部分类的简单案例&mdash;你只需要在一个文件中创建这个类，也许名字叫`MyClass.cs`,对吗？
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/002.md:3]:
```suggestion
如果说你想创建一个名为**MyClass**的类型。在C#中，如果这是一个不涉及部分类的简单案例，你只需要在一个文件中创建这个类，也许名字叫`MyClass.cs`,对吗？
```

File: bite-sized-cpp/posts/cn/002.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+以下是在C++中编写MyClass的常见做法。
+
+* 首先你有一个叫做*头文件*（简称*头*）的东西。那个文件有`.h`扩展名，所以你可以叫它a *.h文件*。它的文件名可能是`MyClass.h`。在这个标题中，你只概述了**MyClass**的基本形态：例如类的名称是什么，类的变量和方法的名称是什么（以及一些其他细节）。
+* 然后你有一个叫做*实现文件*的东西。这个文件有`.cpp`的扩展名，所以你可以叫它a *.cpp文件*。它的文件名可能是`MyClass.cpp`。这就是你把**MyClass**方法的*实现*放在其中的文件；这些方法你只在头文件中*提到*。
+
+虽然，对于非常简单的方法（例如，只返回变量值的方法），您可以选择将该实现放在头中，而不是放在实现文件中。
+
+事实上，您可以选择将*所有*实现放在标题中而不需要实现文件。如果**MyClass**非常简单，使用两个文件似乎有些过分，那么您可能会这么做。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/002.md:12]:
```suggestion
事实上，您可以选择将*所有*实现放在头文件而不需要实现文件。如果**MyClass**非常简单，使用两个文件似乎有些过分，那么您可能会这么做。
```

File: bite-sized-cpp/posts/cn/002.md (hunk 1 above)
Comment (stevewhims) [reply bite-sized-cpp/posts/cn/002.md:3]: Thank you! :)

File: bite-sized-cpp/posts/cn/004.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# #4. 点和箭头
+
+这次让我们看一点语法。假设您有一个名为*myRecording*的变量，类型为**Recording**。首先，这里是使用该变量的常见C#方法。
+
+```csharp
+myRecording.ArtistName = "Wolfgang Amadeus Mozart"; // C#
+```
+我们知道这叫做“点入”变量以访问其成员。上面的示例为*myRecording*的**ArtistName**属性指定了一个Unicode字符串文字。这里有同样的东西移植到现代C++中。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/004.md:8]:
```suggestion
我们知道使用变量+点（或叫做“点入”变量）来访问其成员。上面的示例为*myRecording*的**ArtistName**属性指定了一个Unicode字符串文字。这是移植到现代 C++ 的相同内容。
```

File: bite-sized-cpp/posts/cn/004.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+```cpp
+myRecording.ArtistName(L"Wolfgang Amadeus Mozart"); // C++
+```
+
+我们还在关注*myRecording*。但由于属性不是C++语言的一部分，我们必须使用函数来实现同样的想法。因此C#**ArtistName**string属性变成了C++**ArtistName**取字符串的函数。当您从C#移植到C++/WinRT时，用括号替换等号将成为您的第二天性。此外，ANSI（而非Unicode）字符串是C++中的默认字符串。所以我们需要在字符串文本前面加上'L'，告诉编译器我们指的是Unicode字符串。这是另一个简单的机械替换。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/004.md:14]:
```suggestion
我们还在谈*myRecording*+点。但由于属性不是C++语言的一部分，我们必须使用函数来实现同样的想法。因此C#**ArtistName**string属性变成了C++**ArtistName**取字符串的函数。当您从C#移植到C++/WinRT时，用括号替换等号将成为您的第二天性。此外，ANSI（而非Unicode）字符串是C++中的默认字符串。所以我们需要在字符串文本前面加上'L'，告诉编译器我们指的是Unicode字符串。这是另一个简单的机械替换。
```

File: bite-sized-cpp/posts/cn/004.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+老实说，C#**ArtistName**string属性变成了*2个*C++**ArtistName**函数。除了上面显示的一个（接受字符串但不返回任何内容）之外，还有一个不接受任何内容但返回字符串。他们是二传手和接球手。从技术上讲，在C++中，修改数据成员值的函数称为*更改器*。返回数据成员值的函数称为*访问器*。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/004.md:16]:
```suggestion
老实说，C#**ArtistName**string属性变成了*2个*C++**ArtistName**函数。除了上面显示的一个（接受字符串但不返回任何内容）之外，还有一个不接受任何内容但返回字符串。他们是修改（Setter）和访问（Getter）。从技术上讲，在C++中，修改数据成员值的函数称为*更改器*。返回数据成员值的函数称为*访问器*。
```

File: bite-sized-cpp/posts/cn/004.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# #4. 点和箭头
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/004.md:1]:
```suggestion
# #4. 点和箭头（.和->）
```

File: bite-sized-cpp/posts/cn/005.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# #5. 点和双冒号
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/005.md:1]:
```suggestion
# #5. 点和双冒号 （.和::）
```

File: bite-sized-cpp/posts/cn/006.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+# #6. 包括头文件（第1部分）
+
+还记得[#2.头文件（.h文件）](002.md)中的**MyClass**的C++版本吗？我们说过（通常）你在头中描述函数，但你把它们的实现放在实现文件中。为什么要这么麻烦地把事情分开？
+
+假设你是一个软件供应商，拥有一个数学函数库，其中包含了你的专有算法（这是你昂贵的研发成果，来之不易）。您已经在一个名为**MathLibClass**的类中实现了这些函数。传统上，您会向客户提供头文件、静态链接库和[`.dll`](https://docs.microsoft.com/windows/win32/dlls/dynamic-link-libraries).头文件描述了函数，因此您的客户知道每个函数的名称、返回的内容以及使用的参数。在他们的项目中，您的客户链接静态链接库，这样构建过程就可以从这些函数名转换为`.dll`中的入口点。而你的客户会打包`.dll`到其应用程序，因为其中包含函数的实现&amp;mdash；但只能以机器可读（而不是人类可读）的形式。您已经为客户提供了SDK，但您的知识产权（IP）完好无损。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/006.md:5]:
```suggestion
假设你是一个拥有数学函数库的软件供应商，，其中包含了你的专有算法（这是你昂贵的研发成果，来之不易）。您已经在一个名为**MathLibClass**的类中实现了这些函数。传统上，您会向客户提供头文件、静态链接库和[`.dll`](https://docs.microsoft.com/windows/win32/dlls/dynamic-link-libraries).头文件描述了函数，因此您的客户知道每个函数的名称、返回的内容以及使用的参数。在他们的项目中，您的客户链接静态链接库，这样构建过程就可以从这些函数名转换为`.dll`中的入口点。而你的客户会打包`.dll`到其应用程序，因为其中包含函数的实现&amp;mdash；但只能以机器可读（而不是人类可读）的形式。您已经为客户提供了SDK，但您的知识产权（IP）完好无损。
```

File: bite-sized-cpp/posts/cn/006.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+不过，在C#中，任何人都可以相对容易地在您发布的程序集中查看您的源代码（IP），除非您混淆了它。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/006.md:7]:
```suggestion
不过，在C#中，任何人都可以相对容易地在您发布的程序集中查看您的源代码（IP），除非您使用工具混淆（obfuscate）了它。
```

File: bite-sized-cpp/posts/cn/007.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+# #7. 包括头文件（第2部分）
+
+这一次，我们来讨论如何包含C++/WinRT头文件。我们的场景是，我们想写一些异步代码。在这里，我们不会详细讨论如何编写异步代码（我们将把它留到下次使用）。在这里，我们将只关注我们想要的类型，以及什么头允许我们访问该类型。
+
+所以，正如你所知，异步Windows API自Windows 8以来一直风靡一时。在[C++/WinRT](https://docs.microsoft.com/windows/uwp/cpp-and-winrt-apis/)项目中（无论是UWP还是Windows App SDK），这意味着您通常需要[**Windows::Foundation::iSyncAction**](https://docs.microsoft.com/uwp/api/windows.foundation.iasyncaction)接口。这是在执行异步时一直弹出的类型。
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/007.md:5]:
```suggestion
所以，正如你所知，异步Windows API自Windows 8以来一直风靡一时。在[C++/WinRT](https://docs.microsoft.com/windows/uwp/cpp-and-winrt-apis/)项目中（无论是UWP还是Windows App SDK），这意味着您通常需要[**Windows::Foundation::ISyncAction**](https://docs.microsoft.com/uwp/api/windows.foundation.iasyncaction)接口。这是在执行异步时一直弹出的类型。
```

File: bite-sized-cpp/posts/cn/007.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+但是，在使用**Windows::Foundation::iSyncAction**类型之前，您需要让项目了解它。为此，需要包含一个头文件。在C++/WinRT中，包含的头文件是以所需类型的名称空间*命名的**iSyncAction**位于**Windows::Foundation**命名空间中，因此我们执行以下操作：
```
Comment (licanhua) [inline bite-sized-cpp/posts/cn/007.md:7]:
```suggestion
但是，在使用**Windows::Foundation::ISyncAction**类型之前，您需要让项目了解它。为此，需要包含一个头文件。在C++/WinRT中，包含的头文件是以所需类型的名称空间*命名的**ISyncAction**位于**Windows::Foundation**命名空间中，因此我们执行以下操作：
```

**Comment by stevewhims** [comment]
Thank you so much, @Osirisoo0O and @licanhua! :)

-Steve

**Pull request merged**